aten/src/THC/THCAllocator.cpp

```diff
@@ -4,21 +4,30 @@ THCIpcDeleter::~THCIpcDeleter() {
   int prev_device;
   THCudaCheck(cudaGetDevice(&prev_device));
   THCudaCheck(cudaSetDevice(device_));
-  THCudaCheck(cudaIpcCloseMemHandle(data_));
   THCudaCheck(cudaSetDevice(prev_device));
```

Comment:
You don't need the `cudaSetDevice` here anymore. In fact, you don't need this destructor at all.